features/filesystem/storagelite/source/StorageLite.h

```diff
@@ -0,0 +1,559 @@
+     *
+     * @returns STORAGELITE_SUCCESS           Success.
+     *          STORAGELITE_NOT_FOUND         File was not found.
+     *          STORAGELITE_READ_ERROR        Physical error reading data.
+     *          STORAGELITE_DATA_CORRUPT      Data is corrupt.
+     *          STORAGELITE_BAD_VALUE         Bad value in any of the parameters.
+     */
+    int remove(const uint8_t *file_name, size_t file_name_size);
+
+    /**
+     * @brief Returns name of first file and a search handle for next iterations.
+     *
+     * @param[in]  file_name             File name buffer.
+     * @param[in]  file_name_buf_size    File name buffer size.
+     * @param[out] actual_file_name_size Actual file name size.
+     * @param[out] handle                Iteration handle.
+     *
+     * @returns STORAGELITE_SUCCESS           Success.
+     *          STORAGELITE_NOT_FOUND         File was not found (no existing files).
+     *          STORAGELITE_READ_ERROR        Physical error reading data.
+     *          STORAGELITE_DATA_CORRUPT      Data is corrupt.
+     *          STORAGELITE_BAD_VALUE         Bad value in any of the parameters.
+     *          STORAGELITE_BUFF_TOO_SMALL    Not enough memory in user buffer.
+     */
+    int get_first_file(uint8_t *file_name, size_t file_name_buf_size,
+                       size_t& actual_file_name_size, uint32_t& handle);
+
+    /**
+     * @brief Returns name of next file given iteration handle (handle is updated).
+     *        Function return is STORAGELITE_NOT_FOUND at the end of the search.
+     *
+     * @param[in]  file_name             File name buffer.
+     * @param[in]  file_name_buf_size    File name buffer size.
+     * @param[out] actual_file_name_size Actual file name size.
+     * @param[out] handle                Iteration handle.
+     *
+     * @returns STORAGELITE_SUCCESS           Success.
+     *          STORAGELITE_NOT_FOUND         File was not found (past last file).
+     *          STORAGELITE_READ_ERROR        Physical error reading data.
+     *          STORAGELITE_DATA_CORRUPT      Data is corrupt.
+     *          STORAGELITE_BAD_VALUE         Bad value in any of the parameters.
+     *          STORAGELITE_BUFF_TOO_SMALL    Not enough memory in user buffer.
+     */
+    int get_next_file(uint8_t *file_name, size_t file_name_buf_size,
+                      size_t& actual_file_name_size, uint32_t& handle);
+
+    /**
+     * @brief Initializes StorageLite component.
+     *
+     * @param[in]  bd                     Underlying block device.
+     * @param[in]  max_files              Maximal number of files.
+     *
+     * @returns STORAGELITE_SUCCESS       Initialization completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data (on recovery).
+     */
+    int init(BlockDevice *bd, size_t max_files = STORAGELITE_MAX_FILES);
```

Comment:
This followed the FileSystem base class. I agree it is awakward and it made me add some code piruets, specially at init and reset. However, changing this would be a breaking change for all current file systems (FAT, LittleFS), and would require changes in legacy code (PAL code in client is one example). So, regretfully had to follow current base class.

Reply:
Code pirouettes is an amazing word choice. Also I understand the problem now. :+1: 

I agree this should stay as is. We should see if we can change this in the FileSystem API in the future.